Update bool value entries for host config (#23985)
dotnet/AspNetCore.Docs: aspnetcore/fundamentals/host/generic-host.md

## Settings for all app types

> This section lists host settings that apply to both HTTP and non-HTTP workloads. By default, environment variables used to configure these settings can have a `DOTNET_` or `ASPNETCORE_` prefix, which appear in the following configuration for the `{PREFIX_}` placeholder. For more information, see the [Default builder settings](#default-builder-settings) section.

Reviewer: ```suggestion
This section lists host settings that apply to both HTTP and non-HTTP workloads. By default, environment variables used to configure these settings can have a `DOTNET_` or `ASPNETCORE_` prefix, which appear in the following list of settings as the `{PREFIX_}` placeholder. For more information, see the [Default builder settings](#default-builder-settings) section.
```

By [default](xref:fundamentals/configuration/index#default), *appsettings.json* and *appsettings.{Environment}.json* are reloaded when the file changes. To disable this reload behavior in ASP.NET Core 5.0 or later, set the `hostBuilder:reloadConfigOnChange` key to `false`.

**Key**: `hostBuilder:reloadConfigOnChange`  
**Type**: `bool` (`true` or `false`)  
**Default**: `true`  
**Command-line argument**: `hostBuilder:reloadConfigOnChange`  
**Environment variable**: `{PREFIX_}hostBuilder:reloadConfigOnChange`

> [!WARNING]
> The colon (`:`) separator doesn't work with environment variable hierarchical keys on all platforms. For more information, see [Environment variables](xref:fundamentals/configuration/index#environment-variables).

## Settings for web apps

> Some host settings apply only to HTTP workloads. By default, environment variables used to configure these settings can have a `DOTNET_` or `ASPNETCORE_` prefix, which appear in the following configuration for the `{PREFIX_}` placeholder.

Reviewer: ```suggestion
Some host settings apply only to HTTP workloads. By default, environment variables used to configure these settings can have a `DOTNET_` or `ASPNETCORE_` prefix, which appear in the following list of settings as the `{PREFIX_}` placeholder.
```

When enabled, or when the environment is `Development`, the app captures detailed errors.

**Key**: `detailedErrors`  
**Type**: `bool` (`true`/`1` or `false`/`0`)  
**Default**: `false`  

> **Environment variable**: `<PREFIX_>_DETAILEDERRORS`

Reviewer: There's a few cases of this `<PREFIX_>_` double `_` in here. Do you want to fix them while you're in here, or shall I patch them up in a separate PR?

Author: I'll fix them ... and I think we've (or I've) been settling more on braces for these.
